src/dos/dos_files.cpp

```diff
@@ -385,16 +385,17 @@ bool DOS_Rename(const char* const oldname, const char* const newname)
 	return false;
 }
 
-bool DOS_FindFirst(const char *search, uint16_t attr, bool fcb_findfirst)
+bool DOS_FindFirst(const char* search, FatAttributeFlags attr, bool fcb_findfirst)
 {
-	LOG(LOG_FILES,LOG_NORMAL)("file search attributes %X name %s",attr,search);
+	LOG(LOG_FILES, LOG_NORMAL)
+	("file search attributes %X name %s", attr._data, search);
 	DOS_DTA dta(dos.dta());
 	uint8_t drive;char fullsearch[DOS_PATHLENGTH];
 	char dir[DOS_PATHLENGTH];char pattern[DOS_PATHLENGTH];
 	size_t len = strlen(search);
 
 	const bool is_root = (len > 2) && (search[len - 2] == ':') &&
-	                     (attr == DOS_ATTR_VOLUME);
+	                     (attr == FatAttributeVolume);
```

Comment:
Why can't we just check for `attr.volume`?

Reply:
Seems to be safe. Changed, but in a separate commit, for better bisectability in case of problems.